src/main/java/ome/security/basic/BasicSecuritySystem.java

```diff
@@ -374,6 +377,23 @@ public void loadEventContext(boolean isReadOnly, boolean isClose) {
             ec = (EventContext) ste.sessionContext;
         }
 
+        // Ensure that sudoer has name loaded for SimpleEventContext.copy
+        if (ec instanceof SessionContext) {
+            final ome.model.meta.Session session = ((SessionContext) ec).getSession();
+            Experimenter sudoer = session.getSudoer();
+            if (!(sudoer == null || sudoer.isLoaded())) {
+                final Long sudoerId = sudoer.getId();
+                final IQuery iQuery = sf.getQueryService();
+                final String hql = "SELECT omeName FROM Experimenter WHERE id = :id";
+                final Parameters params = new Parameters().addId(sudoerId);
+                final List<Object[]> result = iQuery.projection(hql, params);
+                final String sudoerName = (String) result.get(0)[0];
+                sudoer = new Experimenter(sudoerId, true);
+                sudoer.setOmeName(sudoerName);
```

Comment:
This is technically a no-no. By saying, "true" you're saying that this object is hydrated but only setting one field. If this were to leak into another context and be saved, the DB would throw due to missing fields.

It might be safer to follow the idiom on line 406 and `admin.userProxy(sudoerId);`. That will then be (de-)serialized as with the rest of the system.